src/deepsparse/transformers/eval_downstream.py

```diff
@@ -365,6 +367,48 @@ def conll2003_eval(args):
     return conll2003_metrics
 
 
+def go_emotions_eval(args):
+    # load go_emotions validation dataset and eval tool
+    go_emotions = load_dataset("go_emotions")["validation"]
+    num_labels = 28
+
+    # load pipeline
+    text_classify = Pipeline.create(
+        task="text-classification",
+        model_path=args.onnx_filepath,
+        engine_type=args.engine,
+        num_cores=args.num_cores,
+        sequence_length=args.max_sequence_length,
+        top_k=num_labels,
+    )
+    print(f"Engine info: {text_classify.engine}")
+
+    go_emotions_metrics = PrecisionRecallF1(id_to_label=text_classify.config.id2label)
+    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
```

Comment:
I get an key error on this line with the following command:

```bash
deepsparse.transformers.eval_downstream \
"zoo:nlp/sentiment_analysis/bert-base/pytorch/huggingface/sst2/12layer_pruned80_quant-none-vnni" \
--dataset go_emotions
```

Traceback:
```bash
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 612, in <module>
    main()
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 608, in main
    _main(args)
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 593, in _main
    metrics = SUPPORTED_DATASETS[dataset](args)
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 387, in go_emotions_eval
    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
  File "/home/XXXXX/projects/deepsparse/src/deepsparse/transformers/eval_downstream.py", line 387, in <listcomp>
    ordered_labels = [text_classify.config.id2label[idx] for idx in range(num_labels)]
KeyError: 2

Process finished with exit code 1
```

this is tested on current branch with the following transformers:
```bash
Name: transformers
Version: 4.23.1
Summary: State-of-the-art Machine Learning for JAX, PyTorch and TensorFlow
Home-page: https://github.com/huggingface/transformers
Author: The Hugging Face team (past and future) with the help of all our contributors (https://github.com/huggingface/transformers/graphs/contributors)
Author-email: transformers@huggingface.co
License: Apache
Location: /home/XXXXX/virtual_environments/deepsparse3.8/lib/python3.8/site-packages
Requires: filelock, huggingface-hub, numpy, packaging, pyyaml, regex, requests, tokenizers, tqdm
Required-by: farm-haystack, sentence-transformers
```

Could we re-verify before landing?

Reply:
The model you are using is trained for sst2, not go emotions so is failing due to a label mismatch